fault-proof/src/proposer.rs

```diff
@@ -470,133 +462,37 @@ where
-            .enumerate()
-            .map(|(idx, proof)| {
-                proof.ok_or_else(|| anyhow::anyhow!("missing proof for range index {idx}"))
-            })
-            .collect::<Result<Vec<_>>>()?;
-
-        let boot_infos = boot_infos
-            .into_iter()
-            .enumerate()
-            .map(|(idx, boot)| {
-                boot.ok_or_else(|| anyhow::anyhow!("missing boot info for range index {idx}"))
-            })
-            .collect::<Result<Vec<_>>>()?;
-
-        let latest_l1_head = boot_infos.last().context("No boot infos generated")?.l1Head;
-
-        let headers = match fetcher.get_header_preimages(&boot_infos, latest_l1_head).await {
-            Ok(headers) => headers,
-            Err(e) => {
-                tracing::error!("Failed to get header preimages: {e}");
-                bail!("Failed to get header preimages: {e}");
-            }
-        };
-
-        tracing::info!("Preparing Stdin for Agg Proof");
-        let sp1_stdin = match get_agg_proof_stdin(
-            proofs,
-            boot_infos,
-            headers,
-            &self.prover.range_vk,
-            latest_l1_head,
-            self.signer.address(),
-        ) {
-            Ok(s) => s,
-            Err(e) => {
-                tracing::error!("Failed to get agg proof stdin: {e}");
-                bail!("Failed to get agg proof stdin: {e}");
-            }
-        };
-
-        let tx_hash = self.agg_proof_request(&game, &sp1_stdin).await?;
-
-        Ok((tx_hash, total_instruction_cycles, total_sp1_gas))
-    }
-
-    async fn range_proof_stdin(
-        &self,
-        start_block: u64,
-        end_block: u64,
-        l1_head_hash: B256,
-    ) -> Result<SP1Stdin> {
         let host_args = self
             .host
-            .fetch(start_block, end_block, Some(l1_head_hash), self.config.safe_db_fallback)
+            .fetch(
+                l2_block_number.to::<u64>() - self.config.proposal_interval_in_blocks,
+                l2_block_number.to::<u64>(),
+                Some(l1_head_hash.into()),
+                self.config.safe_db_fallback,
+            )
             .await
             .context("Failed to get host CLI args")?;
 
-        let witness_data = self.host.run(&host_args).await?;
-
-        let sp1_stdin = match self.host.witness_generator().get_sp1_stdin(witness_data) {
-            Ok(stdin) => stdin,
-            Err(e) => {
-                tracing::error!("Failed to get proof stdin: {}", e);
-                return Err(anyhow::anyhow!("Failed to get proof stdin: {}", e));
-            }
-        };
-
-        Ok(sp1_stdin)
-    }
+        let host_clone = self.host.clone();
+        let sp1_stdin = tokio::task::spawn_blocking(move || {
+            let rt = tokio::runtime::Handle::current();
+            rt.block_on(async move {
+                let witness_data = host_clone.run(&host_args).await?;
+                host_clone.witness_generator().get_sp1_stdin(witness_data)
```

Comment:
Would it be better to log each errors before returning it?
```suggestion
                let witness_data = host_clone.run(&host_args).await.inspect_err(|e| tracing::error!("Failed to generate witness: {e}"))?;
                let sp1_stdin = host_clone.witness_generator().get_sp1_stdin(witness_data).inspect_err(|e| tracing::error!("Failed to get proof stdin: {e}"))?;
                Ok::<SP1Stdin, anyhow::Error>(sp1_stdin)
```